test_general/api/test_app.py

```diff
@@ -553,6 +575,72 @@ def json(self):
         return self.body
 
 
+def call_api_using_test_client(
+    request_url: str,
+    api_key: str,
+    filename: str,
+    file,
+    content_type: str,
+    client: TestClient,
+    **partition_kwargs,
+) -> str:
+    """Exact copy of call_api from call_api.py, but with the test client parameter added."""
+    headers = {"unstructured-api-key": api_key}
+
+    response = client.post(
+        MAIN_API_ROUTE,
+        files={"files": (filename, file, content_type)},
+        data=partition_kwargs,
+        headers=headers,
+    )
+
+    if response.status_code != 200:
+        detail = response.json().get("detail") or response.text
+        raise HTTPException(status_code=response.status_code, detail=detail)
+
+    return response.text
+
+
+def test_parallel_mode_preserves_uniqueness_of_hashes_when_assembling_pages_splits(monkeypatch):
+    monkeypatch.setenv("UNSTRUCTURED_PARALLEL_MODE_URL", "unused")
+    monkeypatch.setenv("UNSTRUCTURED_PARALLEL_MODE_ENABLED", "true")
+    monkeypatch.setenv("UNSTRUCTURED_PARALLEL_MODE_SPLIT_SIZE", "1")
+
+    client = TestClient(app)
+    monkeypatch.setattr(
+        general,
+        "call_api",
+        lambda *args, **kwargs: call_api_using_test_client(*args, client=client, **kwargs),
+    )
+
+    # -- there are 3 pages identical pages in this pdf --
+    test_file = Path("sample-docs") / "DA-1p-with-duplicate-pages.pdf"
+    response = client.post(
```

Comment:
during testing we want to use TestClient from FastAPI, but the way splitting is implemented relies on sending API request with PDF fragments to itself (recursively, kind of). So to inject the test client I tried to mock it using `call_api_using_test_client`, expecting that everything will work fine. Unfortunately this test fails, and page_numbers are not assigned correctly...